Odata-docs/webapi-8/getting-started.md

```diff
@@ -0,0 +1,381 @@
+        {
+            var item = customers.SingleOrDefault(d => d.Id.Equals(key));
+
+            if (item == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(item);
+        }
+    }
+}
+```
+In the controller, we defined a `List<Customer>` collection which has five `Customer` objects. This will serve as our in-memory data source for the OData service.
+
+We also defined two overloads of the `Get` method. The first overload does not accept any arguments while the second one accepts a single integer argument. In addition, the first overload returns a list of `Customer` objects while the second one returns a single `Customer` object. These two methods will handle HTTP GET requests from the client. Both overloads are decorated with the `EnableQuery` attribute. This attribute is responsible for applying the query options that are passed in the query string.
+
+## Build the Edm model and configure the service
+Replace the code in the _Startup.cs_ class with the following:
+```csharp
+namespace Lab01
+{
+    using Lab01.Models;
+    using Microsoft.AspNetCore.Builder;
+    using Microsoft.AspNetCore.OData;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.OData.ModelBuilder;
+
+    public class Startup
+    {
+        public void ConfigureServices(IServiceCollection services)
+        {
+            var modelBuilder = new ODataConventionModelBuilder();
+            modelBuilder.EntityType<Order>();
+            modelBuilder.EntitySet<Customer>("Customers");
+
+            services.AddControllers().AddOData(
+                options => options.Select().Filter().OrderBy().Expand().Count().SetMaxTop(null).AddRouteComponents(
+                    "odata",
+                    modelBuilder.GetEdmModel()));
+        }
+
+        public void Configure(IApplicationBuilder app)
+        {
+            app.UseRouting();
+
+            app.UseEndpoints(endpoints => endpoints.MapControllers());
+        }
+    }
+}
+```
+In the _Startup.cs_ class, we define the Edm model. As part of the model, we register `Order` as an entity type and `Customers` as an entity set - `Customer` entity type also gets registered as a result. In this case, the `ODataConventionalModelBuilder` is being used to build the Edm model.
+
+We then proceed to add essential OData services by calling the `AddOData` method, in the process enabling different OData query capabilities - `$select`, `$filter`, `$orderby`, `$expand`, `count`, `$top` and `$skip`. The `AddRouteComponents` method is used to register a route passing along the Edm model to associate it with.
+
+The final step in configuring the service is adding the two middlewares responsible for matching incoming HTTP requests and dispatching those requests to the application's executable endpoints. We do this by calling `UseRouting` and `UseEndpoints` - the former adds route matching to the middleware pipeline while the latter adds endpoint execution.
```

Comment:
I think it would be great to make terms like `endpoint execution` links that point to the relevant pages in the ASP.NET Core docs, cause it's not clear what "endpoint execution" means.

Reply:
@habbes I have added the links.